etcdserver/v3_server.go

```diff
@@ -100,13 +100,17 @@ func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeRe
 	chk := func(ai *auth.AuthInfo) error {
 		return s.authStore.IsRangePermitted(ai, r.Key, r.RangeEnd)
 	}
-
-	get := func() { resp, err = s.applyV3Base.Range(nil, r) }
-	if serr := s.doSerialize(ctx, chk, get); serr != nil {
-		err = serr
+	get := func() error {
+		resp, err = s.applyV3Base.Range(nil, r)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
While I agree that we should not ignore error, why don't you just `return err`? Later we are checking whether it's nil.

Reply:
your advice is ok